pkg/nodeset/v1alpha1/types.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+type NodeClassResourceType string
+
+const (
+	NodeClassResourceFile      NodeClassResourceType = "File"
+	NodeClassResourceReference NodeClassResourceType = "Reference"
+)
+
+type NodeClassResource struct {
+	// Type is the type of the resource
+	Type NodeClassResourceType `json:"type,omitempty"`
+
+	// Name is the name of the resource
+	Name string `json:"name,omitempty"`
+
+	// For File type
+	// Path is the path of the file
+	// +optional
+	Path string `json:"path,omitempty"`
+	// Owner is the owner of the file
+	// +optional
+	Owner string `json:"owner,omityempty"`
+	// Permission is the permission of the file
+	// +optional
+	Permission string `json:"permission,omityempty"`
+	// Template is the template or content of the file
```

Comment:
Template is the counterpart of Reference, isn't it? So eigher Template or Reference is set. Can you make that clearer? Also move the empty line above template, not below.